langchain/document_loaders/youtube.ts

```diff
@@ -0,0 +1,109 @@
+import type YouTubeTranscriptT from "youtube-transcript";
+import type ytdlT from "ytdl-core";
+import { Document } from "../document";
+import { BaseDocumentLoader } from "./base";
+
+let ytdl: typeof ytdlT | null = null;
+let YouTubeTranscript: typeof YouTubeTranscriptT | null = null;
+
+try {
+  // eslint-disable-next-line global-require,import/no-extraneous-dependencies
+  ytdl = require("ytdl-core");
+  // eslint-disable-next-line global-require,import/no-extraneous-dependencies
+  YouTubeTranscript = require("youtube-transcript");
+} catch {
+  // ignore error, will be throw in constructor
+}
+
+export class YouTubeLoader extends BaseDocumentLoader {
+  private videoId: string;
```

Comment:
fyi these aren't truly private, they can still be accessed if you want to. See https://www.typescriptlang.org/docs/handbook/2/classes.html#caveats. If you want them to be truly impossible to access outside the class then this is what you want instead https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields